lib/msal-node/src/network/HttpClient.ts

```diff
@@ -72,6 +78,7 @@ export class HttpClient implements INetworkModule {
                 this.proxyUrl,
                 HttpMethod.POST,
                 options,
+                this.customAgent as http.Agent,
```

Comment:
What if the customer configured https.Agent? Should you not throw an exception with a good error message explaining how to configure these?

Reply:
I made some improvements to address this.

Take a look at this [stack overflow](https://stackoverflow.com/questions/64117101/what-is-the-role-of-https-agent-in-node) post. 

> https.Agent internally calls and constructs with http.Agent

I interpret this as meaning either one can be used. Look at my improvements and let me know your thoughts.